### PR TITLE
feat(crd): use collectorset-controller service to assign resources to a collector

### PR DESCRIPTION
**Pull request by andrewrynhard**
This is a WIP.

See #30.

### CODE REVIEW

**Comment by woz5999** [comment]
This has like 4600 changed files. In its current state, it's unreviewable (and barely loads in my browser). Is this intentional? If this includes upstream updates or something, can we break those into a separate PR?

**Comment by andrewrynhard** [comment]
@woz5999 The changes are required in this PR to get the correct version of kubernetes client, api, and apimachinery to make the CRD possible. Sorry, not much I can do about it.

**Comment by andrewrynhard** [comment]
Looks like the "Files changed" tab shows all the relevant code changes, and doesn't load the vendor diff.

**Comment by woz5999** [comment]
What I'm saying is, put in a specific PR that ONLY updates the k8s client, api, apimachinery version and then a separate one with the specific CRD changes.

**Comment by andrewrynhard** [comment]
@woz5999 I refactored into multiple commits. You can see the subset of changes that are CRD specific by clicking on the commit.

**Comment by woz5999** [comment]
much better

File: pkg/argus.go [hunk 1]
```diff
@@ -142,7 +137,7 @@ func (a *Argus) Watch() {
 		_, controller := cache.NewInformer(
 			watchlist,
 			w.ObjType(),
-			time.Second*0,
+			time.Minute*10,
```
Comment (andrewrynhard) [inline pkg/argus.go]: resync every 10 minutes in case someone manually removes a resource from their LogicMonitor account.

**Pull request merged**